compiler/src/codegen/compcore.re

```diff
@@ -3562,10 +3628,6 @@ let compile_wasm_module = (~env=?, ~name=?, prog) => {
   );
   validate_module(~name?, wasm_mod);
 
-  switch (Config.profile^) {
-  | Some(Release) => Optimize_mod.optimize(wasm_mod)
-  | None => ()
-  };
```

Comment:
This change is to address a Binaryen limitation. I'm not sure that we want to merge with it.

Reply:
Essentially, if optimizations are run, data segments can be removed. Since there is no way to get the name of a segment, the linker is unable to match `memory.init` instructions to the segment they refer to. We get around this by just skipping the optimizations here and relying on the names being the same as the indices at link time, and optimizing the fully linked module at the end. It appears that we get slightly better results by including this optimization here.

Reply:
I opened https://github.com/WebAssembly/binaryen/issues/5984 to track the update needed in Binaryen.